papers/extract.py

```diff
@@ -22,21 +20,55 @@
 # PDF parsing / crossref requests
 # ===============================
 
+def createrandname(path, ext):
```

Comment:
There is already the `tempfile` module, in particular `tempfile.mkstemp`

Reply:
Cool I wasn't aware of this!